keylime/common/algorithms.py

```diff
@@ -12,13 +12,34 @@ def is_accepted(algorithm: str, accepted: List[Any]) -> bool:
     return algorithm in accepted
 
 
+def _hashit(algorithm: str, data: bytes) -> bytes:
+    if algorithm == "sha1":
```

Comment:
One step in the right direction should use `Hash.SHA1`.

But ideally there will be a hidden table with a mapping between the hash name, the hash ID and the hash function, so adding or removing a hash family is editing this table.  The rest of the operation are like `if algorithm in self.supported_hash`, or `return self.supported_hash[algorithm](data).digest()`

Reply:
Again, this is a consequence of having to extract the function from the class to be able to call it from the constructor. I didn't like it either, but if you can fix it in another way...